spec/requests/api/v1/digital_objects_spec.rb

```diff
@@ -118,18 +120,71 @@
+      it 'returns 204' do
+        expect(publication_adapter).to receive(:unpublish).with(authorized_publish_target, digital_object_matcher).and_return([true, []])
+        delete "/api/v1/digital_objects/#{authorized_object.uid}"
         expect(response.status).to be 204
         get "/api/v1/digital_objects/#{authorized_object.uid}"
         expect(response.status).to be 404
       end
-      xit "return a single Digital Object with the expected fields" do
-        expect(JSON.parse(response.body)).to be_blank
+      it 'unpublishes the object on delete' do
+        expect(publication_adapter).to receive(:unpublish).with(authorized_publish_target, digital_object_matcher).and_return([true, []])
+        delete "/api/v1/digital_objects/#{authorized_object.uid}"
+      end
+      it "return blank response entity" do
+        expect(publication_adapter).to receive(:unpublish).with(authorized_publish_target, digital_object_matcher).and_return([true, []])
+        delete "/api/v1/digital_objects/#{authorized_object.uid}"
+        expect(response.body).to be_blank
+      end
+    end
+  end
+
+  describe 'POST /api/v1/digital_objects/:id/publish' do
+    include_examples 'requires user to have correct permissions' do
+      let(:request) do
+        post "/api/v1/digital_objects/#{authorized_object.uid}/publish"
+      end
+    end
+
+    context "logged in" do
+      let(:publication_adapter) { instance_double(Hyacinth::Adapters::PublicationAdapter::Abstract) }
+      let(:configured_publication_adapter) { Hyacinth.config.publication_adapter }
+      before do
+        configured_publication_adapter
+        Hyacinth.config.publication_adapter = publication_adapter
+        sign_in_project_contributor to: [:publish_objects, :read_objects], project: authorized_project
+        allow(publication_adapter).to receive(:update_doi).with(digital_object_matcher, anything)
+      end
+      after do
+        Hyacinth.config.publication_adapter = configured_publication_adapter
+      end
+
+      it 'returns 200' do
+        expect(publication_adapter).to receive(:publish).with(authorized_publish_target, digital_object_matcher, anything).and_return([true, []])
+        post "/api/v1/digital_objects/#{authorized_object.uid}/publish"
+        expect(response.status).to be 200
+      end
+      it 'unpublishes the object on delete' do
+        expect(publication_adapter).to receive(:publish).with(authorized_publish_target, digital_object_matcher, anything).and_return([true, []])
+        post "/api/v1/digital_objects/#{authorized_object.uid}/publish"
+      end
+      it "return json entity" do
+        expect(publication_adapter).to receive(:publish).with(authorized_publish_target, digital_object_matcher, anything).and_return([true, []])
+        post "/api/v1/digital_objects/#{authorized_object.uid}/publish"
+        new_object = JSON.parse(response.body)
+        expect(new_object['projects'].first).to include('string_key' => authorized_project.string_key)
```

Comment:
Should this be checking that the publish target on the object was set correctly?